internal/workloadstatus/errortypes.go

```diff
@@ -0,0 +1,91 @@
+package workloadstatus
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrNoPodsDeployed    = errors.New("no pods deployed")
+	ErrDaemonSetNotFound = errors.New("DaemonSet is not yet created")
+	ErrDaemonSetFetching = errors.New("failed to get DaemonSet")
+)
+
+type PodIsNotScheduledError struct {
+	Message string
+}
+
+func (pns *PodIsNotScheduledError) Error() string {
+	return fmt.Sprintf("Pod is not scheduled: %s", pns.Message)
+}
+
+func IsPodIsNotScheduledError(err error) bool {
+	var pns *PodIsNotScheduledError
+	return errors.As(err, &pns)
+}
+
+type PodIsPendingError struct {
+	ContainerName string
+	Reason        string
+	Message       string
+}
+
+func (p PodIsPendingError) Error() string {
+	return fmt.Sprintf("Pod is in pending state: reason: %s, message: %s", p.Reason, p.Message)
+}
+
+func IsPodIsPendingError(err error) bool {
+	var pipe *PodIsPendingError
+	return errors.As(err, &pipe)
+}
+
+type PodIsFailingError struct {
+	Message string
+}
+
+func (pfe *PodIsFailingError) Error() string {
+	return fmt.Sprintf("Pod has failed: %s", pfe.Message)
+}
+
+func IsPodFailedError(err error) bool {
+	var pfe *PodIsFailingError
+	return errors.As(err, &pfe)
+}
+
+type FailedToListReplicaSetError struct {
+	Message string
```

Comment:
the only usages of this field store content created by err.Error() here. why not store the error here and therefor persist the original error type?